File: lat2/5.py
import pandas as pd
import mysql.connector
import numpy as np
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query1 = "SELECT*FROM order_detail"
query2 = "SELECT*FROM payment_detail"
query3 = "SELECT*FROM sku_detail"
df1 = pd.read_sql(query1, connection)
df2 = pd.read_sql(query2, connection)
df3 = pd.read_sql(query3, connection)
connection.close()

df = pd.merge(df1, df2, left_on='payment_id', right_on='id', how='inner')
df = df[df['is_valid'] == 1]
df['order_date'] = pd.to_datetime(df['order_date'])
df = df[df['order_date'].dt.year == 2022]
print(df)
hasil = df.groupby('payment_method')['id_x'].nunique().reset_index(name='total').sort_values('total', ascending=False).reset_index(drop=True)
top5_payment  = hasil.head(5)
print(top5_payment)
print('\n\n\n\n')

df2 = pd.merge(df, df3, left_on='sku_id', right_on='id', how='inner')
df2 = df2[df2['payment_method'].isin(top5_payment['payment_method'])]
print(df2)
hasil_akhir = df2.groupby(['category','payment_method'])['id_x'].nunique().reset_index(name='total').sort_values('category', ascending=True).reset_index(drop=True)
print(hasil_akhir)

File: lat2/5.r
library(RMySQL)
library(dplyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query1 <- "SELECT * FROM order_detail"
df1 <- dbGetQuery(con, query1)
query2 <- "SELECT * FROM payment_detail"
df2 <- dbGetQuery(con, query2)
query3 <- "SELECT * FROM sku_detail"
df3 <- dbGetQuery(con, query3)
dbDisconnect(con)

df <- inner_join(df1, df2, by = c('payment_id' = 'id'))
df$order_date <- ymd(df$order_date) 
df <- filter(df, is_valid == 1, year(order_date) == 2022)
print(head(df))
hasil <- df %>% group_by(payment_method) %>% summarise(total = n_distinct(id)) %>% arrange(desc(total))
top5 <- head(hasil, 5)
print(top5)



df_lain <- merge(df, df3, by.x = 'sku_id', by.y = 'id')
df_lain <- df_lain[df_lain$payment_method %in% top5$payment_method, ]
print(df_lain)
hasil <- df_lain %>% group_by(category, payment_method) %>% summarise(total = n_distinct(id)) %>% arrange(category)
print(head(hasil))
print(tail(hasil))